novelwriter/gui/doceditor.py

```diff
@@ -3220,3 +3585,74 @@ def updateMainCount(self, count: int, selection: bool) -> None:
         else:
             text = self._trMainCount.format("0", "+0")
         self.wordsText.setText(text)
+
+    def updateVimModeStatusBar(self, modeName: str) -> None:
+        """Update the vim Mode status information."""
+        self.vimStatus.setText(self.tr(modeName))
+
+
+class VimState:
+    """Minimal Vim state machine."""
+
+    __slots__ = (
+        "PREFIX_KEYS",
+        "SUFFIX_KEYS",
+        "VISUAL_PREFIX_KEYS",
+        "_internalClipboard",
+        "_mode",
+        "_normalCommand",
+        "_visualCommand",
+    )
+
+    def __init__(self) -> None:
+        self.PREFIX_KEYS = ["d", "y", "g", "z"]
+        self.VISUAL_PREFIX_KEYS = ["g"]
+        self.SUFFIX_KEYS = ["w"]
```

Comment:
As far as I can tell, these lists can just be changed to strings. A string is also classified as a sequence in Python. As long as you only check a single character, there's no need to make them lists.

```suggestion
        self.PREFIX_KEYS = "dygz"
        self.VISUAL_PREFIX_KEYS = "g"
        self.SUFFIX_KEYS = "w"
```